x-pack/plugins/ml/public/file_datavisualizer/components/edit_flyout/edit_flyout.js

```diff
@@ -0,0 +1,141 @@
+
+import {
+  EuiFlyout,
+  EuiFlyoutHeader,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiTitle,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+  EuiButtonEmpty,
+} from '@elastic/eui';
+
+import { Overrides } from './overrides';
+
+export class EditFlyout extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isFlyoutVisible: false,
+    };
+
+    this.applyOverrides = () => {};
+  }
+
+  componentDidMount() {
+    if (typeof this.props.setShowFunction === 'function') {
+      this.props.setShowFunction(this.showFlyout);
+    }
+  }
+
+  componentWillUnmount() {
+    if (typeof this.props.unsetShowFunction === 'function') {
+      this.props.unsetShowFunction();
+    }
+  }
+
+  closeFlyout = () => {
+    this.setState({ isFlyoutVisible: false });
+  }
+
+  showFlyout = () => {
+    this.setState({ isFlyoutVisible: true });
+  }
+
+  apply = () => {
+    this.applyOverrides();
+    this.closeFlyout();
+  }
+
+  setApplyOverrides = (applyOverrides) => {
+    this.applyOverrides = applyOverrides;
+  }
+  unsetApplyOverrides = () => {
+    this.applyOverrides = () => {};
+  }
+
+  render() {
+    let flyout;
+
+    if (this.state.isFlyoutVisible) {
+      const {
+        setOverrides,
+        overrides,
+        originalSettings,
+        fields,
+      } = this.props;
+
+      flyout = (
+        <EuiFlyout
+          // ownFocus
+          onClose={this.closeFlyout}
+          size="m"
+        >
+          <EuiFlyoutHeader>
+            <EuiTitle>
+              <h2>
+                Override settings
+              </h2>
+            </EuiTitle>
+          </EuiFlyoutHeader>
+          <EuiFlyoutBody>
+
+            <Overrides
+              setOverrides={setOverrides}
+              overrides={overrides}
+              originalSettings={originalSettings}
+              setApplyOverrides={this.setApplyOverrides}
+              fields={fields}
+            />
+
+            {/* <EuiTabbedContent
+              tabs={tabs}
+              initialSelectedTab={tabs[0]}
+              onTabClick={() => { }}
+            /> */}
+
+          </EuiFlyoutBody>
+          <EuiFlyoutFooter>
+            <EuiFlexGroup justifyContent="spaceBetween">
+              <EuiFlexItem grow={false}>
+                <EuiButtonEmpty
+                  iconType="cross"
+                  onClick={this.closeFlyout}
+                  flush="left"
+                >
+                  Close
+                </EuiButtonEmpty>
+              </EuiFlexItem>
+              <EuiFlexItem grow={false}>
+                <EuiButton
+                  onClick={this.apply}
+                  fill
+                  // isDisabled={(isValidJobDetails === false) || (isValidJobCustomUrls === false)}
+                >
+                  Apply
+                </EuiButton>
+              </EuiFlexItem>
+            </EuiFlexGroup>
+          </EuiFlyoutFooter>
+        </EuiFlyout>
+      );
+    }
+    return (
+      <div>
+        {flyout}
```

Comment:
Suggestion: Could be changed to use inline conditional rendering (https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator) to make the use of let+if unncessary, something like:

```
<div>
  {this.state.isFlyoutVisible && (
    <EuiFlyout
      ...
    </EuiFlyout>
  )}
</div>
```